web/src/engine/predictive-text/worker-thread/src/main/predict-helpers.ts

```diff
@@ -590,6 +593,35 @@ export function processSimilarity(
   });
 }
 
+/**
+ * This function may be used to prevent auto-selection/auto-correct from applying in
+ * unexpected ways.  For example, when typing numbers in English, we don't expect
+ * '5' to auto-correct to '5th' just because there are no pure-number entries in
+ * the lexicon rooted on '5'.
+ * @param correction 
+ * @returns 
+ */
+export function correctionValidForAutoSelect(correction: string) {
+  let chars = [...correction];
+  
+  // If the _correction_ - the actual, existing text - does not include any letters,
+  // then predictions built upon it should not be considered valid for auto-correction.
+  for(let c of chars) {
+    // Found even one letter?  We'll consider it valid.
+    switch(searchForProperty(c.codePointAt(0))) {
```

Comment:
This is not going to be sufficient for languages that use digits as letters. I am not excited about adding code that makes assumptions about language properties -- we've tried to avoid this in Keyman because we just don't know all the use cases. Is there another way we can approach this where this knowledge is embedded in the lexical model rather than in KeymanWeb itself?

Reply:
So... maybe some sort of 'regex' for acceptable word forms to auto-correct?

We could take that further and do this PR's current approach as a baseline, but then _additionally_ test with the regex to overrule that behavior for such languages.

Reply:
Something like that would be okay. Allowing override is probably all we need -- the default will probably be fine for most languages.

Reply:
Noted as #12952.